src/v/security/krb5.h

```diff
@@ -10,82 +10,100 @@
 
 #pragma once
 
+#include "outcome.h"
+#include "seastarx.h"
+
+#include <seastar/core/sstring.hh>
+
 #include <krb5/krb5.h>
 
+#include <optional>
 #include <string_view>
 
 namespace security::krb5 {
 
-class krb5_context_view {
+class context_view {
 public:
-    explicit krb5_context_view(const ::krb5_context* ctx)
+    explicit context_view(::krb5_context ctx)
       : _ctx(ctx) {}
 
-    ::krb5_context operator*() const { return *_ctx; }
-    explicit operator const ::krb5_context*() { return _ctx; }
+    ::krb5_context operator*() const { return _ctx; }
```

Comment:
I think this is tripping me up a bit. There are lots of pointers here. `::krb5_context` is already a `::_krb5_context*`, which is probably to make the API types opaque. I.e., we can't embed a `::_krb5_context`.


The `context_view` isn't a "container" of the underlying `::krb5_context`; getting it via `operator*()` doesn't feel right, and also allows mutation via the "view".

I feel like there should be one of these (non-explicit conversion to pointer to const underlying): 
```c++
operator const ::_krb5_context*() const()
```
But that doesn't allow interop with the c API due to const-correctness. If we ignore this issue, would it be clearer to have a conversion to pointer to non-const underlying?
```c++
operator ::_krb5_context() const()
```

Should it be explicit?

Should it be called something else, like `for_read_only_c_api()`?

Reply:
Yeah maybe using an `operator*` is incorrect and instead should just be `operator ::krb5_context()`.

> The `context_view` isn't a "container".... allows mutation via the "view"

I understand what you're saying but I need to be able to access the `::krb5_context` for the C api but also maintain some semblance of ownership (via `context`).

The issue, frankly, is `default_realm` whose destructor requires a `::krb5_context` to simply free memory.  If I didn't need to somehow store a copy of the context for that purpose, I wouldn't need the view.